Source/WebKit2/NetworkProcess/soup/NetworkProcessSoup.cpp

```diff
@@ -50,9 +52,12 @@ void NetworkProcess::userPreferredLanguagesChanged(const Vector<String>& languag
     SoupNetworkSession::defaultSession().setAcceptLanguages(languages);
 }
 
-void NetworkProcess::setProxies(WebCore::SessionID, const Vector<WebCore::Proxy>& proxies)
+void NetworkProcess::setProxies(WebCore::SessionID sessionID, const Vector<WebCore::Proxy>& proxies)
 {
-    SoupNetworkSession::defaultSession().setProxies(proxies);
+    if (auto *storageSession = NetworkStorageSession::storageSession(sessionID))
```

Comment:
What is this fixing? Isn't sessionID always the default session ID here? Or are you using private browsing mode? I Think proxy settings should be applied to all possible sessions, because it's a global configuration.